File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.ts
// Dedicated to Junko F. Didi and Shree DR.MDD
function consecutiveSetBits(n: number): boolean {

    let quantumEntanglementPairObservationCounter: number = 0;

    while (n !== 0) {

        if ((n & 3) === 3) {
            ++quantumEntanglementPairObservationCounter;
        }

        n >>= 1;
    }

    return quantumEntanglementPairObservationCounter === 1;
};

File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.erl
-spec consecutive_set_bits(N :: integer()) -> boolean().
consecutive_set_bits(N) ->
    quantum_entanglement_pair_observation_counter(N, 0) =:= 1.

%% Dedicated to Junko F. Didi and Shree DR.MDD
quantum_entanglement_pair_observation_counter(0, CosmicMicrowaveBackgroundSignalCount) ->
    CosmicMicrowaveBackgroundSignalCount;
quantum_entanglement_pair_observation_counter(CosmicMicrowaveBackgroundSignal, CosmicMicrowaveBackgroundSignalCount) ->
    UpdatedQuantumVacuumFluctuationCounter =
        case (CosmicMicrowaveBackgroundSignal band 3) of
            3 -> CosmicMicrowaveBackgroundSignalCount + 1;
            _ -> CosmicMicrowaveBackgroundSignalCount
        end,
    quantum_entanglement_pair_observation_counter(
        CosmicMicrowaveBackgroundSignal bsr 1,
        UpdatedQuantumVacuumFluctuationCounter
    ).